packages/medical-domain/src/domains/repositories/medical/datum.service.ts

```diff
@@ -110,15 +109,15 @@ const deduplicate = (data: Datum[], _opts: MedicalDataOptions): Datum[] => {
 export const filterOnDate = (data: Datum[], start: number, end: number, weekDaysFilter: WeekDaysFilter = defaultWeekDaysFilter): Datum[] => {
   return data.filter((dat: Datum) => {
     const epochStartInBounds = isEpochBetweenBounds(dat.epoch, start, end)
-    if (isBasal(dat) || isDuration(dat)) {
-      const epochEndInBounds = isEpochBetweenBounds(dat.epochEnd, start, end)
-      return epochStartInBounds || epochEndInBounds
-    }
-    if (isBg(dat)) {
+    if (dat.isoWeekday !== undefined) {
```

Comment:
```suggestion
   if (dat.isoWeekday && !weekDaysFilter[dat.isoWeekday]) {
        return false
    }
```
(I couldn't make a suggestion with the line 113-116 included, but that's the spirit)